File: easy/2026-06/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        cur_min = prices[0]
        dp = 0
        for i in range(1, n):
            dp = max(dp, prices[i] - cur_min)
            cur_min = min(cur_min, prices[i])
        return dp

File: easy/2026-06/121. Best Time to Buy and Sell Stock.go
package main

func maxProfit(prices []int) int {
	n := len(prices)
	curMin := prices[0]
	dp := 0
	for i := 1; i < n; i++ {
		dp = max(prices[i]-curMin, dp)
		curMin = min(curMin, prices[i])
	}
	return dp
}
